assignment1/assignment1.py

```diff
@@ -1 +1,173 @@
+    except ZeroDivisionError:
+        return "You can't divide by 0!"
+    except TypeError:
+        return "You can't multiply those values!"
+
+# Example usage:
+print(calc(5, 0, "divide"))
+print(calc(5, "hello", "multiply"))
+
+# Task 4: Data Type Conversion
+
+def data_type_conversion(value, data_name):
+     try:
+        match data_name:
+            case "int":
+                return int(value)
+            case "float":
+                return float(value)
+            case "str":
+                return str(value)
+            case _:
+                return "Invalid data type!"
+     except ValueError:
+         return f"You can't convert {value} into a {data_name}"
+
+print(data_type_conversion("string", "float"))
+print(data_type_conversion("5", "int"))
+
+# Task 5: Grading System, Using *args
+
+def grade(*args):
+    try:
+        if not args:
+            return "Invalid data was provided."
+
+        average = sum(args) / len(args)
+
+        match average:
+            case avg if avg >= 90:
+                return "A"
+            case avg if avg >= 80:
+                return "B"
+            case avg if avg >= 70:
+                return "C"
+            case avg if avg >= 60:
+                return "D"
+            case _:
+                return "F"
+    except (TypeError, ValueError):
+        return "Invalid data was provided."
+
+
+# Example usage:
+print(grade(85, 90, 78))
+print(grade(50, "hello", 70))
+
+# Task 6: Use a For Loop with a Range
+
+def repeat(string, count):
+    result = ""
+    for i in range(count):
+        result += string
+    return result
+
+print(repeat("florida", 5))
+
+# Task 7: Student Scores, Using **kwargs
+
+def student_scores(mode, **kwargs):
+    try:
+        if not kwargs:
+            return "No scores provided."
+
+        match mode:
+            case "best":
+                return max(kwargs, key=kwargs.get)  # Returns the student with the highest score
+            case "mean":
+                return sum(kwargs.values()) / len(kwargs)  # Returns the average score
+            case _:
+                return "Invalid mode. Use 'best' or 'mean'."
+    except:
+        return "An error occurred."
+
+
+print(student_scores("best", Alice=90, Bob=85, Charlie=92))
+
+# Task 8: Titleize, with String and List Operations
+def titleize(title):
+    words = title.split()
+    small_words = ("a", "on", "an", "the", "of", "and", "is", "in")
+    result = []
+
+    for i, word in enumerate(words):
+        if i == 0 or word not in small_words:
+            result.append(word.capitalize())
+        else:
+            result.append(word)
+
+    return " ".join(result)
+
+print(titleize("hello world and my beautiful husband and kids"))
+
+def hangman(secret, guess):
+    all_letters = list(secret)
+
+    result = []
+    for letter in all_letters:
+        if letter in guess:
+            result.append(letter)
+        else:
+            result.append("_")
+
+    return " ".join(result)
```

Comment:
Extra space is added causing tests to fail
```suggestion
    return "".join(result)
```